### PR TITLE
use ConfirmIdentityDialog instead of ReceiveKeyDialog

### PR DESCRIPTION
**Pull request by agrajaghh**
We currently have two different dialogs to confirm a changed Identity:
`ConfirmIdentityDialog` and `ReceiveKeyDialog`. This PR gets rid of the latter. `ConfirmIdentityDialog` seems to handle all message confilcts of a contact, while you have to confirm every single message with `ReceiveKeyDialog`.

This also switches to AppCompatDialog (see #3222)

### CODE REVIEW

File: src/org/thoughtcrime/securesms/ConfirmIdentityDialog.java [hunk 1]
```diff
@@ -194,20 +194,17 @@ public void onClick(DialogInterface dialog, int which) {
...
+    private VerifySpan(Context context, IdentityKeyMismatch mismatch) {
+      this.context  = context;
+      this.mismatch = mismatch;
     }
 
     @Override
     public void onClick(View widget) {
       Intent intent = new Intent(context, VerifyIdentityActivity.class);
       intent.putExtra("recipient", mismatch.getRecipientId());
-      intent.putExtra("master_secret", masterSecret);
```
Comment (agrajaghh) [inline src/org/thoughtcrime/securesms/ConfirmIdentityDialog.java]: just to be sure: there aren't any `putExtra("master_secret", masterSecret)` calls after the routing activity change supposed to be in the code if the activity extends `PassphraseRequiredActionBarActivity`, right?

File: src/org/thoughtcrime/securesms/ConversationItem.java [hunk 2]
```diff
@@ -369,10 +370,21 @@ private void setContactPhotoForRecipient(final Recipient recipient) {
...
+  private IdentityKeyMismatch getKeyMismatch(final MessageRecord record) {
+    if (record.isIdentityMismatchFailure()) {
+      for (final IdentityKeyMismatch mismatch : record.getIdentityKeyMismatches()) {
+        if (mismatch.getRecipientId() == record.getIndividualRecipient().getRecipientId()) {
+          return mismatch;
+        }
+      }
+    }
+    return null;
+  }
```
Comment (agrajaghh) [inline src/org/thoughtcrime/securesms/ConversationItem.java]: I copied this method from https://github.com/WhisperSystems/TextSecure/blob/master/src/org/thoughtcrime/securesms/MessageRecipientListItem.java#L142. Where do you think its best to put it? Just making it `public static` there didn't seem like a good idea...

**Comment by agrajaghh** [comment]
This fixes btw a crash which can happen if you confirm messages in the wrong order

File: src/org/thoughtcrime/securesms/ConfirmIdentityDialog.java (hunk 1 above)
Comment (mcginty) [inline src/org/thoughtcrime/securesms/ConfirmIdentityDialog.java]: that's correct

**Comment by moxie0** [comment]
In 2.17.0, thanks!

**Pull request closed without merging**